contracts/smart-account/libs/AddressResolver.sol

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+interface IAddressResolver {
+    function getAddressForCounterFactualAccount(
+        address moduleSetupContract,
+        bytes calldata moduleSetupData,
+        uint256 index
+    ) external view returns (address _account);
+
+    function getAddressForCounterFactualAccount(
+        address _owner,
+        uint256 _index
+    ) external view returns (address _account);
+}
+
+interface IEcdsaRegistryModule {
+    function initForSmartAccount(address eoaOwner) external returns (address);
+}
+
+interface ISmartAccount {
+    function getImplementation()
+        external
+        view
+        returns (address _implementation);
+
+    function VERSION() external view returns (string memory);
+}
+
+contract AddressResolver {
+    struct SmartAccountResult {
+        address accountAddress;
+        address factoryAddress;
+        address currentImplementation;
+        string currentVersion;
+        string factoryVersion;
+        uint256 deploymentIndex;
+    }
+
+    address public constant SA_V1_FACTORY =
+        0x000000F9eE1842Bb72F6BBDD75E6D3d4e3e9594C;
+    address public constant SA_V2_FACTORY =
+        0x000000a56Aaca3e9a4C479ea6b6CD0DbcB6634F5;
+    // marked for deletion
+    address public constant SA_V1_IMPLEMENTATION =
+        0x00006B7e42e01957dA540Dc6a8F7C30c4D816af5;
+    // marked for deletion
+    address public constant SA_V2_IMPLEMENTATION =
+        0x0000002512019Dafb59528B82CB92D3c5D2423aC;
+    address public constant ECDSA_REGISTRY_MODULE_ADDRESS =
+        0x0000001c5b32F37F5beA87BDD5374eB2aC54eA8e;
+
+    // Todo Add natspec
+    // Review could pass factory and module address in params
+    function resolveEOAtoAccountAddresses(
```

Comment:
factory, module and setupdata should be passed as parameters here so that we can use them with other modules

Reply:
I feel this is very specific to Biconomy Factories. and only for V1 and V2

like they must have to pass V1 factory address
and they must have to pass V2 factory address
which are fixed.

module address and module setup data can be passed in the method yes. but they will also only work with V2.

hence it's difficult to extend this contract to work with wider use cases and currently specific case (for biconomy, for v1 and v2 only) is enough.

Reply:
got it